srcs/ResponseHandler.cpp

```diff
@@ -178,17 +160,22 @@ void ResponseHandler::extractCGIheaders(const std::string& incoming) {
 }
 
 void ResponseHandler::handleBody(request_s& request) {
-	int		ret = 1024;
 	char	buf[1024];
-	int		fd;
-	int totalreadsize = 0;
+	int		ret = 1024,
+			fd,
+			totalreadsize = 0;
 
 	_body.clear();
-	if (request.status_code == 400) {
+	if (request.status_code >= 400 && request.status_code < 500) {
 		fd = open(request.location->geterrorpage().c_str(), O_RDONLY);
-	}
-	else {
+	} else {
 		fd = generatePage(request);
+		if (fd == -1)
+			fd = open(request.location->geterrorpage().c_str(), O_RDONLY);
+	}
+	if (fd < 0) {
+		_body = "Why are you here? Something obviously went wrong.\n";
```

Comment:
Shouldn't we return a proper error in this case?

Reply:
We are returning an error in this case, but apparently it's impossible to read from the errorpage as well, so this is the body then